plugin-server/src/worker/ingestion/event-pipeline/extractWebVitalsDataStep.ts

```diff
@@ -0,0 +1,111 @@
+import { eventDroppedCounter } from '../../../main/ingestion-queues/metrics'
+import { PreIngestionEvent, RawClickhouseWebVitalsEvent, TimestampFormat } from '../../../types'
+import { castTimestampOrNow } from '../../../utils/utils'
+import { captureIngestionWarning, isNonEmptyString } from '../utils'
+import { EventPipelineRunner } from './runner'
+
+const validWebVitalNames = ['LCP', 'FCP', 'INP', 'CLS']
+
+interface ReceivedWebVitalEvent {
+    name: string
+    value: number
+    // we're not using the rest of the fields
```

Comment:
we're dumping info here e.g. events can be raised not on page navigation
we should probably also have a properties bag so we can filter in future or only store the events related to a  navigation for now